### ISSUE TITLE
[1.18.2] Wasting huge parts of the texture atlas with the electrolytic seperator

### ISSUE THREAD

**Issue by Chaos02** [mekanism/Mekanism#7848]
## The general problem

Hey I've been loving Mekanism for ages and finally I could afford a high tier modern GPU - I've been playing a 300 Mod Modpack and have just been beginning compiling a Sphax patch for it.

Now that I'm getting to a point where I've compiled a notable amount of patches I've been getting the same and same error again, when trying to apply patches that work on their own:

I keep getting "reloading ressource pack failed" in the top right corner and getting booted back to vanilla textures BUT
this is only happening once I seem to cross some threshold. Example:
1. I try out the AE2Things patch by itself - it works
2. I load many of the Patches for my modpack - it works.
3. I add the AE2Things patch on top - I get booted back to vanilla

<details>
   <summary>(expand) This is the exception I could fish out of the (more than 30mill lines long) log:</summary>

```
[17Aug.2023 22:38:22.150] [Render thread/INFO] [net.minecraft.client.Minecraft/]: Caught error loading resourcepacks, removing all selected resourcepacks
java.util.concurrent.CompletionException: net.minecraft.ReportedException: Stitching
    at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315) ~[?:?]
    at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320) ~[?:?]
    at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1770) ~[?:?]
    at java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760) ~[?:?]
    at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[?:?]
    at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[?:?]
    at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[?:?]
    at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[?:?]
    at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[?:?]
Caused by: net.minecraft.ReportedException: Stitching
    at net.minecraft.client.renderer.texture.TextureAtlas.m_118307_(TextureAtlas.java:140) ~[client-1.18.2-20220404.173914-srg.jar%23427!/:?]
    at net.minecraft.client.resources.model.ModelBakery.processLoading(ModelBakery.java:223) ~[client-1.18.2-20220404.173914-srg.jar%23427!/:?]
    at net.minecraftforge.client.model.ForgeModelBakery.<init>(ForgeModelBakery.java:61) ~[forge-1.18.2-40.2.9-universal.jar%23432!/:?]
    at net.minecraft.client.resources.model.ModelManager.m_5944_(ModelManager.java:57) ~[client-1.18.2-20220404.173914-srg.jar%23427!/:?]
    at net.minecraft.client.resources.model.ModelManager.m_5944_(ModelManager.java:20) ~[client-1.18.2-20220404.173914-srg.jar%23427!/:?]
    at net.minecraft.server.packs.resources.SimplePreparableReloadListener.m_10786_(SimplePreparableReloadListener.java:11) ~[client-1.18.2-20220404.173914-srg.jar%23427!/:?]
    at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) ~[?:?]
    ... 6 more
Caused by: net.minecraft.client.renderer.texture.StitcherException: Unable to fit: pneumaticcraft:item/upgrades/charging - size: 16x16 - Maybe try a lower resolution resourcepack?
    at net.minecraft.client.renderer.texture.Stitcher.m_118193_(Stitcher.java:66) ~[client-1.18.2-20220404.173914-srg.jar%23427!/:?]
    at net.minecraft.client.renderer.texture.TextureAtlas.m_118307_(TextureAtlas.java:132) ~[client-1.18.2-20220404.173914-srg.jar%23427!/:?]
    at net.minecraft.client.resources.model.ModelBakery.processLoading(ModelBakery.java:223) ~[client-1.18.2-20220404.173914-srg.jar%23427!/:?]
    at net.minecraftforge.client.model.ForgeModelBakery.<init>(ForgeModelBakery.java:61) ~[forge-1.18.2-40.2.9-universal.jar%23432!/:?]
    at net.minecraft.client.resources.model.ModelManager.m_5944_(ModelManager.java:57) ~[client-1.18.2-20220404.173914-srg.jar%23427!/:?]
    at net.minecraft.client.resources.model.ModelManager.m_5944_(ModelManager.java:20) ~[client-1.18.2-20220404.173914-srg.jar%23427!/:?]
    at net.minecraft.server.packs.resources.SimplePreparableReloadListener.m_10786_(SimplePreparableReloadListener.java:11) ~[client-1.18.2-20220404.173914-srg.jar%23427!/:?]
    at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) ~[?:?]
    ... 6 more

```

</details>

## The part where you come in

It was pointed out to me what could be going wrong when applying these patches. Although I have an RX6800 with 16GB of VRAM, The maximum texture size on many modern GPUs is not *great*, in fact for mine it's "only" 16k *x* 16k.

I appreciate that you've already put good effort into UV unwrapping the digital miner, the modification station and the advanced solar panel, HOWEVER:
![Texture Atlas](https://github.com/mekanism/Mekanism/assets/20928702/6acee683-0460-4ad4-8211-3ef12df01970)

As you can see, the **electrolytic separator** wastes a *ginormous* amount of space on my texture atlas, only helped by the fact that I am trying to use a x256 Ressource pack (which tries to make the block actually be x256).

Please resolve this issue and if possible also dish out a patch for older versions of Minecraft (like 1.18 and up)!
As you can probably think, I'm also contacting the authors of other mods, which are wasting giant spaces on my texture atlas
and am internally downsizing problematic textures

Appreciate all your efforts,
Chaos

### Steps to reproduce

1. use mod in a (now) reasonably sized technic / magic mod pack
2. create a Ressource pack patch for said mod pack
3. see yourself fail

### Minecraft version

1.18.x or earlier (No longer being developed)

### Forge version

not important

### Mekanism version

Older

### Other relevant versions

Mekanism 10.2.5

**Comment by Chaos02** [comment]
Edit: I have browsed the repo for some time and found that you've already fixed this issue in 1.19+ (https://github.com/mekanism/Mekanism/commit/2246511418ceb382f4eddb1b927271a8f54ba3c5)
Still it would be great to have this fix on the main modding version as of now - the sphax pack for Mekanism is under active development so it will probably recieve the changes made.

**Comment by thiakil** [comment]
I don't know that there really is a "main modding version" but even so, we don't support older versions as it's too much work to maintain

If you want to make a pr for it we can merge it, but won't likely be making a release
